addon/components/base/bs-form/element.js

```diff
@@ -928,6 +928,14 @@ export default FormGroup.extend({
    */
   onChange() {},
 
+  /**
+   * Private duplicate of onChange event used for internal state handling between form and it's elements.
+   *
+   * @event _onChange
+   * @private
+   */
+  _onChange() {},
```

Comment:
This feels strange but unless I missed something there isn't any other way than having a duplicated `onChange` event for this internal sate tracking.

Reply:
We could call `resetSubmissionState()` in the form's `change` action, and only have a single action called here. But that would break the state handling if the user uses a custom `onChange` handler on the form element. Not sure if we need to support that case, as the user is then handling the form state by her/himself, so she/he could (should?) be responsible for handling the form submission state as well? 

A bot undecided about this one. We could maybe try the latter first, to simplify the implementation, and see if anyone sees this as an issue?